### PR TITLE
Modify kerberos deployment settings

### PR DESCRIPTION
**Pull request by lipppppp**
The kerberos deployment document exist some problems.
1.New principal should have correct permission to execute actions on HDFS when set the smartsever and smartagent principal.
2.The smart-site.xml should be updated on each agent node.

### CODE REVIEW

File: docs/enable-kerberos.md [hunk 1]
```diff
@@ -75,15 +76,16 @@ kadmind
 
 ### 2.1 Add smartserver Kerberos principal to database and export it to keytab.
 ```
-kadmin.local:addprinc -randkey smartserver/{hostname}
-kadmin.local:xst -k /xxx/xxx/smartserver-{hostname}.keytab smartserver/{hostname}
+kadmin.local:addprinc -randkey {username}/{hostname}
+kadmin.local:xst -k /xxx/xxx/smartserver-{hostname}.keytab {username}/{hostname}
 ```
-Note: replace the hostname with the hostname of the smart server
+**Note:** replace the username with the HDFS user who has the correct permission to execute actions and replace hostname with hostname of the smart server
```
Comment (philo-he) [inline docs/enable-kerberos.md:82]: I am not sure what "HDFS user" means. Is it the user configured in hadoop-env.sh? I forget why we used smartserver or smartagent in previous tests.

Comment (philo-he) [reply docs/enable-kerberos.md:82]: Can the username for smart agent and smart server be the same one here?

File: docs/enable-kerberos.md [hunk 2]
```diff
@@ -75,15 +76,16 @@ kadmind
...
 
-### 2.2 Add smartagent Kerberos principals to database and export it to keytabs. Please create principals for each agent.
+### 2.2 Add smartagent Kerberos principals to database and export it to keytabs. Please create principals for each agent.Then use 'scp' to copy each keytab file to each agent.
```
Comment (philo-he) [inline docs/enable-kerberos.md]: A minor format issue, please keep a space after "." or any other punctuation.

File: docs/enable-kerberos.md (hunk 1 above)
Comment (lipppppp) [reply docs/enable-kerberos.md:82]: The user performing the operation changed to smartagent or smartserver after completing kerberos configuration. But these two users do not have permission to operate on some specific files. “HDFS user” means it is best to have the highest operating authority for HDFS like the user configured in dfs.permissions.superusergroup.  
image

File: docs/enable-kerberos.md (hunk 2 above)
Comment (lipppppp) [reply docs/enable-kerberos.md]: OK, I will add a space after ".".

File: docs/enable-kerberos.md (hunk 1 above)
Comment (lipppppp) [reply docs/enable-kerberos.md:82]: Of course, we can set this "HDFS user" as other users name. But we should give the user proper authority by Ranger or  other authorization management application.

Comment (philo-he) [reply docs/enable-kerberos.md:82]: OK, please also use {username} to replace "smartagent"/"smartserver" for setting the value of _smart.server.kerberos.principal/smart.agent.kerberos.principal_ in the below configuration template of smart-site.xml.

Comment (philo-he) [reply docs/enable-kerberos.md:82]: i.e., line 119, 127

Comment (lipppppp) [reply docs/enable-kerberos.md:82]: Thanks. I have replaced "smartagent"/"smartserver" in the smart-site.xml.

**Comment by philo-he** [comment]
The patch looks good to me. Will commit it and ignore the irrelevant build error. Thanks @lipppppp again!

**Pull request merged**